src/perf/counter/group/reader.cpp

```diff
@@ -49,9 +49,13 @@ namespace group
 
 template <class T>
 Reader<T>::Reader(ExecutionScope scope, bool enable_on_exec)
-: counter_buffer_(requested_group_counters().counters.size() + 1)
+: counter_buffer_(CounterProvider::instance()
```

Comment:
We could trade memory for computation here by making the CounterCollection a member.

Reply:
Since we save the CounterCollection in the child class `Writer` anyways, we should do that here instead and only pay once.